### ISSUE TITLE
cv::moveWindow in 3.1.0 does not respond well to y axis input

### ISSUE THREAD

**Issue by JzHuai0108** [opencv/opencv#6650]
### Please state the information for your system
- OpenCV version: 3.1.0
- Host OS: Linux (Ubuntu 14.04)
- _(if needed)_ Compiler & CMake: GCC 4.8 & CMake 3.2
### In which part of the OpenCV library you got the issue?
- moveWindow in highgui module
### Expected behaviour

move window to the specified x and y coordinates, 
### Actual behaviour

move window properly in x direction, but not working well in y direction. given nonzero y, sometimes no translation in y, sometimes, a few pixels.
### Additional description

This function works properly with OpenCV 2.4.8, though.
### Code example to reproduce the issue / Steps to reproduce the issue

Please try to give a full example which will compile as is.

```
#ifdef OPENCV_3_1_0
#include <opencv2/features2d.hpp>
#include <opencv2/imgcodecs.hpp>
#include <opencv2/opencv.hpp>
#else
#include "opencv2/core/core.hpp"
#include "opencv2/highgui/highgui.hpp"
#include "opencv2/imgproc/imgproc.hpp"
#include "opencv2/features2d/features2d.hpp"
#endif

#include <vector>
#include <iostream>

#include<memory>

using namespace std;
using namespace cv;

int main(int argc, char* argv[])
{
    string folder = "../opencv-3.1.0/samples";
    Mat img1 = imread(folder + "/data/graf1.png", IMREAD_GRAYSCALE);
    Mat img2 = imread(folder + "/data/graf3.png", IMREAD_GRAYSCALE);

    int x = atoi(argv[1]);
    int y = atoi(argv[2]);
    std::cout << "move to x,y "<<x <<" "<< y<<std::endl;

    string winName = "Left";
    cv::namedWindow(winName);
    cv::moveWindow(winName, x, y);

    string winName2 = "Right";
    cv::namedWindow(winName2);
    cv::moveWindow(winName2, x+300, y+500);

    cv::imshow(winName, img1);
    cv::imshow(winName2, img2);
    cv::waitKey(3000);
    return 0;
}
```

**Comment by gsk2286** [comment]
@JzHuai0108
 
I tried to reproduce the issue.But the movewindow works as expected .But for small change in x and y changes it is difficult to identify the translation.Could you please tell me for what x and y values the issue was identified.

**Comment by JzHuai0108** [comment]
You are right. moveWindow does work. I have resolved my problem. The problem is actually caused by calling moveWindow in a multithreaded environment. Several routines may try to moveWindow at the same time. After I make them run sequentially, the problem is gone.

**Comment by Creolophous** [comment]
You can't call in not a multithreaded environment, because you always run in on top window manager (OS GUI) anyway.
We tested the issue and it turned out that window simply can not be drawn in 0, 0 because desktop gui components, such as menu bar and side bars. (On ubuntu you always have it) that's where "the few pixel" problem is coming from. The x pixels is exactly the width of the sidebar and y pixels is the height of menu bar.
In summary, there is nothing wrong with that.
